src/test/jtx/batch.h

```diff
@@ -0,0 +1,185 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2024 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_TEST_JTX_BATCH_H_INCLUDED
+#define RIPPLE_TEST_JTX_BATCH_H_INCLUDED
+
+#include <test/jtx/Account.h>
+#include <test/jtx/Env.h>
+#include <test/jtx/amount.h>
+#include <test/jtx/owners.h>
+#include <test/jtx/tags.h>
+#include <concepts>
+#include <cstdint>
+#include <optional>
+
+namespace ripple {
+namespace test {
+namespace jtx {
+
+/** Batch operations */
+namespace batch {
+
+/** Batch. */
+Json::Value
+batch(
+    jtx::Account const& account,
+    uint32_t seq,
+    STAmount const& fee,
+    std::uint32_t flags);
+
+/** Adds a new Batch Txn on a JTx. */
+class add
+{
+private:
+    Json::Value txn_;
+    std::uint32_t seq_;
+    std::optional<std::uint32_t> ticket_;
+
+public:
+    add(Json::Value const& txn,
+        std::uint32_t const& sequence,
+        std::optional<std::uint32_t> const& ticket = std::nullopt,
+        std::optional<std::uint32_t> const& fee = std::nullopt)
+        : txn_(txn), seq_(sequence), ticket_(ticket)
+    {
+    }
+
+    void
+    operator()(Env&, JTx& jtx) const;
+};
+
+/** Set a batch signature on a JTx. */
+class sig
+{
+public:
+    struct Reg
```

Comment:
Is this supposed to refer to a regular key? Why not just call it `RegularKey`?

Reply:
refactored msig: https://github.com/XRPLF/rippled/pull/5060/commits/ef49db7c5e177747296b02b653c04b60dbfc5518